### PR TITLE
extend vxlanFlow tests (APIv1) by checking rules

### PR DESCRIPTION
**Pull request by andriidovhan**
- extend vxlanFlow tests (APIv1) by checking rules
- improve procedure of finding noviflow switches in vxlan tests

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy [hunk 1]
```diff
@@ -631,7 +629,9 @@ class FlowRulesSpec extends HealthCheckSpecification {
         flowHelper.addFlow(flow)
 
         and: "Reproduce situation when switches have missing rules by deleting flow rules from them"
+        def flowInfoFromDb = database.getFlow(flow.id)
         def involvedSwitches = pathHelper.getInvolvedSwitches(flow.id)*.dpId
+        def transitSwitchIds = (involvedSwitches.size() > 2) ? involvedSwitches[-1..-2] : []
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: You pick neighboring switches on L624. Why do you expect transit switches here?

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy [hunk 2]
```diff
@@ -229,7 +228,9 @@ class SwitchSyncSpec extends BaseSpecification {
         flowHelper.addFlow(flow)
 
         and: "Reproduce situation when switches have missing rules and meters"
+        def flowInfoFromDb = database.getFlow(flow.id)
         def involvedSwitches = pathHelper.getInvolvedSwitches(flow.id)
+        def transitSwitchIds = (involvedSwitches.size() > 2) ? involvedSwitches[-1..-2]*.dpId : []
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: same

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy (hunk 1 above)
Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: the why do you assume 'no transit switches' situations then? (also during `|| true` below)

Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: fixed

Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: also the algorithm of finding not neighbouring noviflow switches was fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy [hunk 3]
```diff
@@ -619,19 +618,24 @@ class FlowRulesSpec extends HealthCheckSpecification {
...
 
-    @Ignore("sync is not working yet, fix in pr2591")
     @Tags(HARDWARE)
     def "Able to synchronize rules for a flow with VXLAN encapsulation"() {
-        given: "Two active Noviflow switches"
-        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find { it.src.noviflow && it.dst.noviflow }
+        given: "Two active not neighboring Noviflow switches"
+        def switchPair = topologyHelper.getAllNotNeighboringSwitchPairs().find { swP ->
+            swP.src.noviflow && swP.dst.noviflow && swP.paths.find { path ->
+                pathHelper.getInvolvedSwitches(path).find { it.noviflow }
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: should be `pathHelper.getInvolvedSwitches(path).every { it.noviflow }` if you want a path with all-novi switches

Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: fix

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy [hunk 4]
```diff
@@ -239,17 +240,16 @@ class IntentionalRerouteSpec extends HealthCheckSpecification {
...
         given: "A vxlan flow"
         def allTraffgenSwitchIds = topology.activeTraffGens*.switchConnected.findAll {
-            it.noviflow
+            it.noviflow && !it.wb5164
         }*.dpId ?: assumeTrue("Should be at least two active traffgens connected to NoviFlow switches", false)
         def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find { swP ->
             allTraffgenSwitchIds.contains(swP.src.dpId) && allTraffgenSwitchIds.contains(swP.dst.dpId) &&
                     swP.paths.findAll { path ->
-                        pathHelper.getInvolvedSwitches(path).find { it.noviflow }
+                        pathHelper.getInvolvedSwitches(path).find { it.noviflow  && !it.wb5164 }
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: every?

Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy]: you are right, thx

**Pull request merged**